meson.build

```diff
@@ -98,19 +103,11 @@ if get_option('default_library') != 'static'
 endif
 
 # Check all compiler flags
-foreach cflag: test_cflags
-  if cpp.has_argument(cflag)
-    common_cflags += cflag
-  endif
-endforeach
+common_cflags += cpp.get_supported_arguments(test_cflags)
 
 # Linker flags
-if host_machine.system() == 'linux'
-  foreach ldflag: [ '-Wl,-Bsymbolic', '-Wl,-z,relro', '-Wl,-z,now', ]
-    if cpp.has_argument(ldflag)
-      common_ldflags += ldflag
-    endif
-  endforeach
+if host_machine.system() == 'linux' and cpp.get_id() == 'gcc'
+  common_ldflags += [ '-Wl,-Bsymbolic', '-Wl,-z,relro', '-Wl,-z,now', ]
```

Comment:
Would you not want these flags with Clang?

Reply:
We cannot test for the presence of these flags, unlike with compiler ones; in theory, this syntax is supported by Clang, but there's really no guarantee that any non-GCC compiler will effectively support it without errors.

Reply:
Okay then. I guess doing it properly would depend on a `meson.get_linker()` API.